url.bs

```diff
@@ -88,6 +88,334 @@ valid input. User agents, especially conformance checkers, are encouraged to rep
+     <a>is special</a>, which would result in the following host portion becoming
+     "<code>exa#mple.org</code>".
+     <p>"<code>https://exa%23mple.org</code>"
+    </div>
+   <td>✅
+  <!-- IPv4 parser -->
+  <tr>
+   <td><dfn>empty-IPv4-part</dfn>
+   <td>
+    <p>Input to the <a>IPv4 parser</a> ends with a U+002E.
+    <p class=example id=example-empty-ipv4-part>"<code>https://127.0.0.1./</code>"
+   <td>❌
+  <tr>
+   <td><dfn>too-many-IPv4-parts</dfn>
+   <td>
+    <p>An <a for=/>IPv4 address</a> consists of 4 parts.
+    <p class=example id=example-too-many-ipv4-parts>"<code>https://1.2.3.4.5/</code>"
+   <td>✅
+  <tr>
+   <td><dfn>non-numeric-IPv4-part</dfn>
+   <td>
+    <p>An IPv4 part was not numeric.
+    <p class=example id=example-non-numeric-ipv4-part>"<code>https://test.42</code>"
+   <td>✅
+  <tr>
+   <td><dfn>non-decimal-IPv4-part</dfn>
+   <td>
+    <p>The IPv4 address contains numbers expressed using hexadecimal or octal digits.
+    <p class=example id=example-non-decimal-ipv4-part>"<code>https://127.0.0x0.1</code>"
+   <td>❌
+  <tr>
+   <td><dfn>IPv4-part-out-of-range</dfn>
+   <td>
+    <p>An IPv4 part exceeds 255.
+    <p class=example id=example-ipv4-part-out-of-range>"<code>https://255.255.4000.1</code>"
+   <td>✅ (only if applicable to the last IPv4 part)
+  <!-- IPv6 parser -->
+  <tr>
+   <td><dfn>invalid-compressed-IPv6-address</dfn>
+   <td>
+    <p>An IPv6 address begins with improper compression.
+    <p class=example id=example-invalid-compressed-ipv6-address>"<code>https://[:1]</code>"
+   <td>✅
+  <tr>
+   <td><dfn>IPv6-too-many-pieces</dfn>
+   <td>
+    <p>An IPv6 address contains more than 8 pieces.
+    <p class=example id=example-ipv6-too-many-pieces>"<code>https://[1:2:3:4:5:6:7:8:9]</code>"
+   <td>✅
+  <tr>
+   <td><dfn>IPv6-multiple-compression</dfn>
+   <td>
+    <p>An IPv6 address is compressed in more than one spot.
+    <p class=example id=example-ipv6-multiple-compression>"<code>https://[1::1::1]</code>"
+   <td>✅
+  <tr>
+   <td><dfn>IPv4-in-IPv6-empty-part</dfn>
+   <td>
+    <p>An IPv6 address that contains an IPv4 address has an empty part in the IPv4 address.
+    <p class=example id=example-ipv4-in-ipv6-empty-part>"<code>https://[ffff::.0.0.1]</code>"
+   <td>✅
+  <tr>
+   <td><dfn>IPv4-in-IPv6-too-many-pieces</dfn>
+   <td>
+    <p>An IPv6 address contains an IPv4 address and the IPv6 address has more than 6 pieces.
+    <p class=example id=example-ipv4-in-ipv6-too-many-pieces>"<code>https://[1:1:1:1:1:1:1:127.0.0.1]</code>"
+   <td>✅
+  <tr>
+   <td><dfn>IPv4-in-IPv6-too-many-parts</dfn>
+   <td>
+    <p>An IPv6 address contains an IPv4 address and the IPv4 address has more than 4 parts.
+    <p class=example id=example-ipv4-in-ipv6-too-many-parts>"<code>https://[ffff::127.0.0.1.2]</code>"
+   <td>✅
+  <tr>
+   <td><dfn>IPv4-in-IPv6-unexpected-code-point</dfn>
+   <td>
+    <p>An IPv6 address contains an IPv4 address of which a part contain a code point other than an
```

Comment:
Should probably read ", of which, a part contains code points other than ASCII digits or is an empty string." or feel free to completely reword it.